### PR TITLE
Support ULID

### PR DESCRIPTION
**Pull request by pbzweihander**
This PR adds support for [ulid](https://github.com/dylanhart/ulid-rs).

### CODE REVIEW

File: utoipa-gen/src/schema_type.rs [hunk 1]
```diff
@@ -372,26 +400,36 @@ pub enum Variant {
...
                 }
+                #[cfg(all(feature = "uuid", not(feature = "ulid")))]
+                {
+                    _format != &"Ulid"
+                }
+                #[cfg(all(not(feature = "uuid"), not(feature = "ulid")))]
+                {
+                    _format != &"Ulid" && _format != &"Ulid"
+                }
             })
```
Comment (juhaku) [inline utoipa-gen/src/schema_type.rs:433]: This seems a bit fishy

**Review by juhaku — CHANGES_REQUESTED** [review]
Seems otherwise good, just a test missing from `schema_derive_test.rs` for the ulid case.

Also perhaps the known format thingy could be bit refactored.
```rust
#[cfg(not(any(feature = "uuid", feature = "ulid")))]
{
  true
}
#[cfg(all(feature = "uuid", not(feature = "ulid")))]
{
  format != "Ulid"
}
#[cfg(all(feature = "ulid", not(feature = "uuid")))]
{
  format != "Uuid"
}
```

File: utoipa-gen/src/schema_type.rs (hunk 1 above)
Comment (pbzweihander) [reply utoipa-gen/src/schema_type.rs]: Can we have a better code for this? I cannot come up with it.

**Comment by pbzweihander** [comment]
Can you help with the test? I'm not sure how to write one

File: utoipa-gen/src/schema_type.rs (hunk 1 above)
Comment (juhaku) [reply utoipa-gen/src/schema_type.rs]: Perhaps we can have this as is, and change it in future.

**Comment by juhaku** [comment]
> Can you help with the test? I'm not sure how to write one

https://github.com/juhaku/utoipa/blob/56b6326e28f0df2a88f5b0572c415d6bd9b8fb1a/utoipa-gen/tests/schema_derive_test.rs#L4707-L4730

Just like in above one but it it should test a type that has the `Ulid` type.

```rust
#[test]
#[cfg(feature = "ullid")]
fn test {
    #[derive(ToSchema)]
    struct Foo {
        value: Ulid,
    }
    
    // ... and so on the assertion
}
```

**Comment by pbzweihander** [comment]
I added test. Thanks!

**Review by juhaku — APPROVED** [review]
Great :trophy:

**Pull request merged**